### PR TITLE
[processor/transform] Fix log accessor bugs

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:**
Some accessors for logs were not returning proper data types, breaking their usage in conditions. This PR fixes those bugs.

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10350

**Testing:** 
Updated/Added unit tests

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

File: processor/transformprocessor/internal/logs/logs.go [hunk 1]
```diff
@@ -299,7 +299,7 @@ func accessDroppedAttributesCount() pathGetSetter {
 func accessFlags() pathGetSetter {
 	return pathGetSetter{
 		getter: func(ctx common.TransformContext) interface{} {
-			return ctx.GetItem().(plog.LogRecord).Flags()
+			return int64(ctx.GetItem().(plog.LogRecord).Flags())
```
Comment (bogdandrutu) [inline processor/transformprocessor/internal/logs/logs.go:302]: Flags is unit32?

Comment (TylerHelmuth) [reply processor/transformprocessor/internal/logs/logs.go:302]: Yes

**Comment by TylerHelmuth** [comment]
https://github.com/orgs/open-telemetry/teams/collector-contrib-approvers please review

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,12 +39,14 @@
 - `googlecloudexporter`: Support writing to multiple GCP projects by setting the `gcp.project.id` resource attribute, and support service account impersonation (#11051)
 - `k8sattributeprocessor`: Add debug logs to help identify missing attributes (#11060)
 
+
```
Comment (dmitryax) [inline CHANGELOG.md:42]:
```suggestion
```

**Review by dmitryax — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**